persist-cli/src/main/java/io/ballerina/persist/nodegenerator/SyntaxTreeGenerator.java

```diff
@@ -108,7 +110,52 @@ public static SyntaxTree createPesistToml(String name) {
         return SyntaxTree.from(textDocument);
     }
 
-    public static HashMap<String, String> readToml(Path configPath, String name) throws BalException {
+    public static HashMap<String, String> populateConfiguration(HashMap<String, String> persistConfig, Path configPath)
```

Comment:
IMO Reading of the Config.toml file and resolving templated configs should be separate functions. You are trying to combine 2 functions inside one